lib/transports/rest.js

```diff
@@ -0,0 +1,386 @@
+'use strict'
+
+const crypto = require('crypto')
+const request = require('request')
+
+const { genAuthSig } = require('../util')
+
+const API_URL = 'https://api.bitfinex.com'
+
+/**
+ * Communicates with v1 of the Bitfinex HTTP API
+ */
+class RESTv1 {
+  constructor (opts = { apiKey: '', apiSecret: '', url: API_URL }) {
+    this._url = opts.url || API_URL
+    this._apiKey = opts.apiKey || ''
+    this._apiSecret = opts.apiSecret || ''
+    this._nonce = Date.now()
+    this._generateNonce = (typeof opts.nonceGenerator === 'function')
+        ? opts.nonceGenerator
+        : () => ++this._nonce
+  }
+
+  make_request (path, params, cb) {
+    if (!this._apiKey || !this._apiSecret) {
+      return cb(new Error('missing api key or secret'))
+    }
+
+    const payload = Object.assign({
+      request: `/v1/${path}`,
+      nonce: JSON.stringify(this._generateNonce())
+    }, params)
+
+    const payloadBase64 = new Buffer(JSON.stringify(payload)).toString('base64')
+    const { sig } = genAuthSig(this._apiSecret, payloadBase64)
+
+    return request({
+      url: `${this._url}/v1/${path}`,
+      method: 'POST',
+      timeout: 15000,
+      headers: {
+        'X-BFX-APIKEY': this._apiKey,
+        'X-BFX-PAYLOAD': payloadBase64,
+        'X-BFX-SIGNATURE': sig
+      }
+    }, (err, response, body) => {
+      if (err || (response.statusCode !== 200 && response.statusCode !== 400)) {
+        return cb(new Error(err != null ? err : response.statusCode))
+      }
+
+      let result
+
+      try {
+        result = JSON.parse(body)
+      } catch (error) {
+        return cb(null, {
+          message: body.toString()
+        })
```

Comment:
wait a second...

given this returns a html cloudflare page...

the error is catched, and then the callback called with no error, but message as prop... this will break other peoples code...

Reply:
This was there before, heh: https://github.com/bitfinexcom/bitfinex-api-node/blob/master/rest.js#L56

I'm returning the error now in all cases